spec/rspec/matchers/built_in/satisfy_spec.rb

```diff
@@ -24,6 +24,30 @@
       end
     end.to fail_with("expected false to satisfy block")
   end
+
+  context "when a custom description is provided" do
+    it "describes itself" do
+      expect(satisfy("be awesome").description).to eq("be awesome")
+    end
+
+    it "passes if block returns true" do
+      expect(true).to satisfy("be true") { |val| val }
+      expect(true).to satisfy("be true") do |val|
+        val
+      end
+    end
+
+    it "fails if block returns false" do
```

Comment:
WDYT of `it "fails with the custom description block returns false"`?  That's an important aspect of the behavior.